src/server/server_family.cc

```diff
@@ -554,6 +550,58 @@ string_view GetRedisMode() {
   return ClusterConfig::IsEnabledOrEmulated() ? "cluster"sv : "standalone"sv;
 }
 
+struct ReplicaOfArgs {
+  bool no_one = false;
+  string_view host;
+  string_view port_sv;
+  uint32_t port;
+  std::optional<SlotRange> slot_range;
+  static OpResult<ReplicaOfArgs> FromParams(string_view host, string_view port,
+                                            std::optional<string_view> slot_start,
+                                            std::optional<string_view> slot_end);
+};
+
+OpResult<ReplicaOfArgs> ReplicaOfArgs::FromParams(string_view host, string_view port,
+                                                  std::optional<string_view> slot_start,
+                                                  std::optional<string_view> slot_end) {
+  ReplicaOfArgs replicaof_args;
+  replicaof_args.host = host;
+  replicaof_args.port_sv = port;
+  if (absl::EqualsIgnoreCase(replicaof_args.host, "no") &&
+      absl::EqualsIgnoreCase(replicaof_args.port_sv, "one")) {
+    replicaof_args.no_one = true;
+    return replicaof_args;
+  }
+
+  if (!absl::SimpleAtoi(replicaof_args.port_sv, &replicaof_args.port) || replicaof_args.port < 1 ||
```

Comment:
If you use a `uint16_t` above you could change the range check to simply be `replicaof_arg.port == 0`

Reply:
I dont see absl::SimpleAtoi impl with int16_t
do you have any utility function that we can use ?

Reply:
You're right. I don't know why absl made that decision, but yeah, we need to use a temporary and check its upper range it looks like :|